internal/tiltfile/tiltfile_state.go

```diff
@@ -1068,8 +1056,20 @@ func (s *tiltfileState) imgTargetsForDependencyIDsHelper(ids []model.TargetID, c
 		}
 		claimStatus[id] = claimPending
 
+		registry := s.defaultRegistryHost
+		if s.orchestrator() == model.OrchestratorK8s && !s.privateRegistry.Empty() {
+			// If we've found a private registry in the cluster at run-time,
+			// use that instead of the one in the tiltfile
+			s.logger.Infof("Auto-detected private registry from environment: %s", s.privateRegistry)
+			registry = s.privateRegistry
+		}
+		refs, err := container.NewRefSet(image.configurationRef, registry)
```

Comment:
would it make more sense to do this substitution during tiltfile execution? So RefSet would go on the dockerImage struct.

The problem with this approach is that if there's an error, you'll lose the stack trace, as in this bug: #2768

so we try hard not to return errors from this function without lots of context about what went wrong and how to fix it.

Reply:
ah gotcha, will do.

Reply:
...hmm wait, i don't know if we can do that -- we don't guarantee that `default_registry` is called at any particular point in the tiltfile, right? so we can't assume that by the time the user calls `docker_build`, we've already set `s.registry`.

Reply:
ah, good point, maybe just add additional context to the error?

Reply:
will do